### PR TITLE
External simulation: Providing information about next tick of MobSim

### LINKED ISSUE #776: EV simulations should provide next tick to `ExtEvDataService` as well
Related: 
- https://github.com/ie3-institute/simonaAPI/issues/170
- https://github.com/ie3-institute/simonaAPI/issues/176
- https://github.com/ie3-institute/MobilitySimulator/issues/254

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #776

Depends on:
- https://github.com/ie3-institute/simonaAPI/pull/171

### CODE REVIEW

File: src/main/scala/edu/ie3/simona/service/ev/ExtEvDataService.scala [hunk 1]
```diff
@@ -142,8 +146,6 @@ class ExtEvDataService(override val scheduler: ActorRef)
     serviceStateData.uuidToActorRef.get(evcs) match {
       case None =>
         // Actor is not registered yet
-        agentToBeRegistered ! RegistrationSuccessfulMessage(self, None)
```
Comment (jo-bao) [inline src/main/scala/edu/ie3/simona/service/ev/ExtEvDataService.scala:145]: Perhaps make a comment as to why the message is being sent later.

**Review by jo-bao — COMMENTED** [review]
As always good work! Maybe you can also update ExtEvSimulationSequence.puml?

**Comment by sebastian-peter** [comment]
`ExtEvSimulationClasses` will be dealt with in https://github.com/ie3-institute/simona/issues/898, `ExtEvSimulationSequence` I'm taking care of myself.

**Review by danielfeismann — DISMISSED** [review]
LGTM but maybe @jo-bao would like to also a view here?

**Review by jo-bao — DISMISSED** [review]
Looks good to me!

**Review by danielfeismann — APPROVED** [review]
LGTM

File: src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala [hunk 2]
```diff
@@ -494,32 +494,39 @@ protected trait EvcsAgentFundamentals
...
-    )
+        /* Update the base state data with the updated state data store */
+        modelBaseStateData.copy(
+          stateDataStore = updatedStateDataStore
+        )
+      } else
+        // Empty arrivals means that there is no data for this EVCS at the current tick,
+        // thus we just return and wait for the next activation
+        modelBaseStateData
+    }
```
Comment (danielfeismann) [inline src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala:529]: This seems to cause an issue
```
18:19:22.854 ERROR e.i.s.a.g.GridAgent$$anonfun$edu$ie3$simona$agent$grid$GridAgent$$initializing$1 - The model EvcsModel(ec3cb2fa-6826-4236-bdea-311ab2e54ce0,EVCS_Bus 17,Interval [0, 2764800],2016-01-01T00:00Z,CosPhiFixed(0.9),22.0 kVA,AC,0.9,1,HOME,false,MAX_POWER,0.2) was not provided with needed EV data.
edu.ie3.simona.exceptions.agent.InconsistentStateException: The model EvcsModel(ec3cb2fa-6826-4236-bdea-311ab2e54ce0,EVCS_Bus 17,Interval [0, 2764800],2016-01-01T00:00Z,CosPhiFixed(0.9),22.0 kVA,AC,0.9,1,HOME,false,MAX_POWER,0.2) was not provided with needed EV data.
	at edu.ie3.simona.agent.participant.evcs.EvcsAgentFundamentals.$anonfun$calculatePowerWithSecondaryDataAndGoToIdle$3(EvcsAgentFundamentals.scala:320)
	at scala.Option.getOrElse(Option.scala:201)
	at edu.ie3.simona.agent.participant.evcs.EvcsAgentFundamentals.calculatePowerWithSecondaryDataAndGoToIdle(EvcsAgentFundamentals.scala:319)
	at edu.ie3.simona.agent.participant.evcs.EvcsAgentFundamentals.calculatePowerWithSecondaryDataAndGoToIdle$(EvcsAgentFundamentals.scala:294)
	at edu.ie3.simona.agent.participant.evcs.EvcsAgent.calculatePowerWithSecondaryDataAndGoToIdle(EvcsAgent.scala:66)
	at edu.ie3.simona.agent.participant.evcs.EvcsAgent.calculatePowerWithSecondaryDataAndGoToIdle(EvcsAgent.scala:66)
	at edu.ie3.simona.agent.participant.ParticipantAgent$$anonfun$5.applyOrElse(ParticipantAgent.scala:461)
	at edu.ie3.simona.agent.participant.ParticipantAgent$$anonfun$5.applyOrElse(ParticipantAgent.scala:412)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:35)
	at org.apache.pekko.actor.FSM.processEvent(FSM.scala:861)
	at org.apache.pekko.actor.FSM.processEvent$(FSM.scala:858)
	at edu.ie3.simona.agent.participant.ParticipantAgent.org$apache$pekko$actor$LoggingFSM$$super$processEvent(ParticipantAgent.scala:91)
	at org.apache.pekko.actor.LoggingFSM.processEvent(FSM.scala:999)
	at org.apache.pekko.actor.LoggingFSM.processEvent$(FSM.scala:981)
	at edu.ie3.simona.agent.participant.ParticipantAgent.processEvent(ParticipantAgent.scala:91)
	at org.apache.pekko.actor.FSM.org$apache$pekko$actor$FSM$$processMsg(FSM.scala:855)
	at org.apache.pekko.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:850)
	at org.apache.pekko.actor.Actor.aroundReceive(Actor.scala:547)
	at org.apache.pekko.actor.Actor.aroundReceive$(Actor.scala:545)
	at edu.ie3.simona.agent.participant.ParticipantAgent.aroundReceive(ParticipantAgent.scala:91)
	at org.apache.pekko.actor.ActorCell.receiveMessage(ActorCell.scala:590)
	at org.apache.pekko.actor.ActorCell.invoke(ActorCell.scala:557)
	at org.apache.pekko.dispatch.Mailbox.processMailbox(Mailbox.scala:272)
	at org.apache.pekko.dispatch.Mailbox.run(Mailbox.scala:233)
	at org.apache.pekko.dispatch.Mailbox.exec(Mailbox.scala:245)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

**Pull request merged**